### PR TITLE
Added new framework for Cli-Tests

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
This PR contains a new framework for cli tests. With this new framework it is possible to test the new cli more intuitive and check different artifacts than the options.

### CODE REVIEW

File: cli/src/test/java/de/jplag/cli/test/CliArg.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package de.jplag.cli.test;
+
+public record CliArg<T>(String name, boolean isPositional) {
+    public static CliArg<String[]> SUBMISSION_DIRECTORIES = new CliArg<>("", true);
```
Comment (Kr0nox) [inline cli/src/test/java/de/jplag/cli/test/CliArg.java:4]: Do we want to have every CLI arg as a constant here? Another option I could imagine working is, generating them from the https://github.com/jplag/JPlag/blob/feature/cliTests/cli/src/main/java/de/jplag/cli/options/CliOptions.java Class and then referencing them by there names in the methods were you now give a CliArg Object. 
I dont really have a preference, just wanted to leave that as an option for discussion here

File: cli/src/test/java/de/jplag/cli/test/CliTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    protected JPlagOptions runCliForOptions(Consumer<CliArgBuilder> additionalOptionsBuilder) throws ExitException, IOException {
+        return runCli(additionalOptionsBuilder).jPlagOptions();
+    }
+
+    protected CliResult runCli(Consumer<CliArgBuilder> additionalOptionsBuilder) throws ExitException, IOException {
+        try {
+            try (MockedStatic<JPlagRunner> runnerMock = Mockito.mockStatic(JPlagRunner.class)) {
+                runnerMock.when(() -> JPlagRunner.runJPlag(Mockito.any())).thenReturn(new JPlagResult(Collections.emptyList(), null, 1, null));
+                try (MockedStatic<OutputFileGenerator> generatorMock = Mockito.mockStatic(OutputFileGenerator.class)) {
```
Comment (Kr0nox) [inline cli/src/test/java/de/jplag/cli/test/CliTest.java:56]: Is there a way to write this without 3 try blocks? Especially when only one of them has a catch

File: cli/src/test/java/de/jplag/cli/test/CliTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    @BeforeEach
+    void setup() {
+        this.initializeParameters(this.args);
+    }
+
+    public void addDefaultParameters() {
+        this.args.with(CliArg.SUBMISSION_DIRECTORIES, new String[] {CURRENT_DIRECTORY});
+    }
+
+    public abstract void initializeParameters(CliArgBuilder args);
```
Comment (Kr0nox) [inline cli/src/test/java/de/jplag/cli/test/CliTest.java:86]: Do we want to leave this abstract or make it have a default implementation where it calls `addDefaultParameters`

File: cli/src/test/java/de/jplag/cli/ClusteringTest.java [hunk 4]
```diff
@@ -4,41 +4,53 @@
...
+    void parseDefaultClustering() throws ExitException, IOException {
+        JPlagOptions options = runCliForOptions();
         assertTrue(options.clusteringOptions().enabled());
     }
 
     @Test
-    void parsePercentilePreProcessor() throws CliException {
-        buildOptionsFromCLI(defaultArguments().clusterPpPercentile(.5));
+    void parsePercentilePreProcessor() throws ExitException, IOException {
+        JPlagOptions options = runCliForOptions(args -> args.with(CliArg.CLUSTER_PP_PERCENTILE, .5));
```
Comment (Kr0nox) [inline cli/src/test/java/de/jplag/cli/ClusteringTest.java:34]: Extract `-5` into a constant, also with the assert below

File: cli/src/test/java/de/jplag/cli/OldNewRootDirectoriesArgumentTest.java [hunk 5]
```diff
@@ -2,54 +2,67 @@
...
 
         assertEquals(0, options.submissionDirectories().size());
         assertEquals(0, options.oldSubmissionDirectories().size());
     }
 
     @Test
-    void testTwoRootDirectoryArguments() throws CliException {
-        buildOptionsFromCLI(arguments().rootDirectory("root1", "root2"));
+    void testTwoRootDirectoryArguments() throws ExitException, IOException {
+        JPlagOptions options = runCliForOptions(args -> args.with(CliArg.SUBMISSION_DIRECTORIES, new String[] {"root1", "root2"}));
```
Comment (Kr0nox) [inline cli/src/test/java/de/jplag/cli/OldNewRootDirectoriesArgumentTest.java:26]: Maybe also extract this array into a variable and then use its length in the assert below
Same with all other arrays in this test class

**Review by tsaglam — CHANGES_REQUESTED** [review]
Just a few issues and one minor sonar issue.

File: report-viewer/package-lock.json
Comment (Kr0nox) [inline report-viewer/package-lock.json:1]: Did you make these changes to the package-lock on purpose? running `npm i` readds the lines removed/changed. 
If this was not on purpose, please revert the changes (you can simply run `npm i` and commit that)

**Review by Kr0nox — CHANGES_REQUESTED** [review]

**Comment by tsaglam** [comment]
There were still conflicts for the package lock, I resolved them by overriding with develop.

**Review by tsaglam — APPROVED** [review]

**Review by Kr0nox — APPROVED** [review]

**Pull request merged**